src/ucp/api/ucp.h

```diff
@@ -3707,6 +3710,12 @@ ucs_status_ptr_t ucp_put_nbx(ucp_ep_h ep, const void *buffer, size_t count,
  *
  * @note A user can use @ref ucp_worker_flush_nb "ucp_worker_flush_nb()"
  * in order to guarantee re-usability of the source address @e buffer.
+ * @note The completion of a get operation signals the local @e buffer holds the
+ * the expected data and that both local @e buffer and remote @remote_addr are
```

Comment:
i'd not mention any guarantee about remote buffer

Reply:
This is something that isn't clear to me. As discussed offline, if the get operation completes successfully and the local buffer is now ready for use, to me that intuitively would mean "the remote buffer use is also completed". Is that not right? What else could make it unsafe to reuse the remote buffer once the get operation is confirmed successful in the local worker (who initiated and confirmed the operation complete)?

Reply:
it sounds a bit odd to mention it, as this process does not own/manage that memory

Reply:
Exactly, plus it's not clear even from a testing standpoint, what are the guarantees that are provided by UCP? If I just wait on the `ucp_get_nbx`, am I safe to destroy everything immediately? If not, I think it's worth documenting exactly what is required to ensure what the user is doing is enough/safe.